jme3-examples/src/main/java/jme3test/TestSceneBuilder.java

```diff
@@ -0,0 +1,1014 @@
+            bullet = new BulletAppState();
+            app.getStateManager().attach(bullet);
+        }
+        return bullet;
+    }
+    private PhysicsSpace getOrCreatePhysics() {
+        return getOrCreateBullet().getPhysicsSpace();
+    }
+
+    /**
+     * Dedicated test application for {@link TestSceneBuilder}.
+     */
+    private static class TestApp extends SimpleApplication {
+
+        private static void start(String[] args) {
+            new TestApp().start();
+        }
+
+        @Override
+        public void simpleInitApp() {
+
+            TestSceneBuilder scene = new TestSceneBuilder(this);
+            scene.configure();
+            scene.baseScene();
+            scene.brightMountainsSun();
+            scene.hardwareProbe();
+            scene.brightMountainsSky();
+            scene.softBloom(b -> b.setGlowFactor(0.1f));
+            scene.scattering();
+            scene.antialiasing();
+            scene.shadowFilter();
+            scene.physics();
+            scene.character();
+
+        }
+
+    }
+
+    /**
+     * Simple first person physical character control.
+     */
+    public class FirstPersonCharacter extends BetterCharacterControl implements AnalogListener {
+
+        private static final String FORWARD = "TestScene_Character_Forward";
+        private static final String BACKWARD = "TestScene_Character_Backward";
+        private static final String LEFT = "TestScene_Character_Left";
+        private static final String RIGHT = "TestScene_Character_Right";
+        private static final String JUMP = "TestScene_Character_Jump";
+
+        private final Vector2f input = new Vector2f();
+        private final Vector3f walk = new Vector3f();
+        private float walkSpeed = 10f;
+        private float strafeSpeed = 7f;
+        private float jumpDelay = 0f;
+        private Trigger forwardTrigger = new KeyTrigger(KeyInput.KEY_W);
+        private Trigger backwardTrigger = new KeyTrigger(KeyInput.KEY_S);
+        private Trigger leftTrigger = new KeyTrigger(KeyInput.KEY_A);
+        private Trigger rightTrigger = new KeyTrigger(KeyInput.KEY_D);
+        private Trigger jumpTrigger = new KeyTrigger(KeyInput.KEY_SPACE);
+
+        /**
+         *
+         * @param radius collision shape radius
+         * @param height collision shape height
+         * @param mass rigid body mass
+         */
+        public FirstPersonCharacter(float radius, float height, float mass) {
+            super(radius, height, mass);
+        }
+
+        @Override
+        public void update(float tpf) {
+            jumpDelay -= tpf;
+            app.getCamera().getDirection(walk);
+            walk.multLocal(input.y * walkSpeed).addLocal(app.getCamera().getLeft().mult(input.x * strafeSpeed)).setY(0f);
+            setWalkDirection(walk);
+            input.set(0f, 0f);
+            super.update(tpf);
+            app.getCamera().setLocation(spatial.getWorldTranslation().add(0f, getFinalHeight() - 2f, 0f));
```

Comment:
`spatial` is a private field in the `BetterCharacterControl `class in the `Minie `library and you should avoid using it directly.